zng/recordval.go

```diff
@@ -235,6 +236,18 @@ func (r *Record) Access(field string) (Value, error) {
 	return r.ValueByColumn(col), nil
 }
 
+func (r *Record) Deref(path field.Path) (Value, error) {
+	var v Value
+	for _, f := range path {
+		var err error
+		v, err = r.Access(f)
+		if err != nil {
+			return Value{}, err
+		}
+	}
+	return v, nil
```

Comment:
```suggestion
	v := r.Value
	for _, f := range path {
		typ := zng.TypeRecordOf(v.Type)
		if typ == nil {
			return Value{}, errors.New("field access on non-record value")
		}
		var err error
		v, err = zng.NewVolatileRecord(typ, v.Bytes).Access(f)
		if err != nil {
			return Value{}, err
		}
	}
	return v, nil
```

Reply:
I don't think we need to do this.  Let's discuss.

Reply:
This speaks to the way we think about zng.Value and the forthcoming changes.

Reply:
My bad, but yuck, this stuff is ancient and needs refactoring.  I think operators should be applied to zng.Value and it would switch inside.  Then, we can transform this pattern into the JIT-CG (just-in-time closure generator) that produces an efficient closure as a value opeator for any top-level schema.